includes/amp-helper-functions.php

```diff
@@ -2068,6 +2068,7 @@ function amp_add_admin_bar_view_link( $wp_admin_bar ) {
  * @return string|null Script hash or null if the sha384 algorithm is not supported.
  */
 function amp_generate_script_hash( $script ) {
+	/** @var false|string */
 	$sha384 = hash( 'sha384', $script, true );
 	if ( false === $sha384 ) {
 		return null;
```

Comment:
Looks like this needs to be further improved to be compatible with PHP 8, which is apparently why PHPStan is now complaining here. According to the docs:

<a href="https://www.php.net/manual/en/function.hash.php#refsect1-function.hash-returnvalues"><img width="1056" alt="image" src="https://user-images.githubusercontent.com/134745/168914752-62d34ef9-bef4-4037-aff3-d5f5fdb8d9de.png"></a>

Apparently there should be a try/catch block for the sake of PHP 8, followed by a check for `false` for PHP<8.

```suggestion
	try {
		$sha384 = hash( 'sha384', $script, true );
	} catch ( ValueError $e ) {
		$sha384 = false;
	}
	if ( false === $sha384 ) {
		return null;
```